crates/primitives/src/env/handler_cfg.rs

```diff
@@ -31,20 +34,32 @@ impl HandlerCfg {
                 let is_optimism = false;
             }
         }
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "bsc")] {
+                let is_bsc = true;
+            } else {
+                let is_bsc = false;
+            }
+        }
         Self {
             spec_id,
             #[cfg(feature = "optimism")]
             is_optimism,
+            #[cfg(feature = "bsc")]
+            is_bsc,
         }
     }
 
     /// Creates new `HandlerCfg` instance with the optimism feature.
     #[cfg(feature = "optimism")]
     pub fn new_with_optimism(spec_id: SpecId, is_optimism: bool) -> Self {
-        Self {
-            spec_id,
-            is_optimism,
-        }
+        Self { spec_id, is_optimism }
+    }
+
+    /// Creates new `HandlerCfg` instance with the bsc feature.
+    #[cfg(feature = "bsc")]
+    pub fn new_with_bsc(spec_id: SpecId, is_bsc: bool) -> Self {
+        Self { spec_id, is_bsc }
```

Comment:
will this have missing field, as there are literal three field, and the second one is `is_optimism`.

Reply:
No.